operators/pkg/instance-controller/container_logic.go

```diff
@@ -192,11 +234,45 @@ func (r *InstanceReconciler) CreateContainerEnvironment(
 	vmStart time.Time) error {
 	ctx := context.TODO()
 
-	service, ingress, err := r.CreateInstanceExpositionEnvironment(ctx, instance, name)
+	service, ingress, urlUUID, err := r.CreateInstanceExpositionEnvironment(ctx, instance, name, true)
 	if err != nil {
 		return err
 	}
 
+	cmName := "filebrowser-config"
+
+	fileManagerPort := int32(8080)
+	mountPath := "/mydrive"
+
+	cmData := make(map[string]string)
+	cmData["port"] = fmt.Sprintf("%d", fileManagerPort)
+	cmData["baseURL"] = "/" + urlUUID + "/files"
+	cmData["address"] = ""
+	cmData["log"] = "stdout"
+	cmData["database"] = "/tmp/database.db"
+	cmData["root"] = mountPath
+	cmData["noauth"] = strconv.FormatBool(true)
+
+	cm := v1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cmName,
+			Namespace: namespace,
+		},
+		Data:      cmData,
+		Immutable: pointer.Bool(true),
+	}
+
+	if _, err := ctrl.CreateOrUpdate(ctx, r.Client, &cm, func() error {
```

Comment:
Since you are configuring the ConfigMap as immutable, it becomes useless to leverage the CreateOrUpdate approach. I would just stick to the classical Create function, discarding the error in case the object already exists.